src/Features/Core/Portable/ConvertLinq/AbstractConvertLinqQueryToLinqMethodProvider.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.Simplification;
+
+namespace Microsoft.CodeAnalysis.ConvertLinq
+{
+    internal abstract class AbstractConvertLinqQueryToLinqMethodProvider<TQueryExpression> : CodeRefactoringProvider
```

Comment:
are we going to add vb one in future?

Reply:
yes

---
In reply to: [176886991](https://github.com/dotnet/roslyn/pull/25362#discussion_r176886991) [](ancestors = 176886991)